lib/spack/spack/config.py

```diff
@@ -227,6 +235,11 @@ def get_section_filename(self, section: str) -> str:
 
     def get_section(self, section: str) -> Optional[YamlConfigDict]:
         """Returns the data associated with a given section"""
+        if not self.readable:
+            # warn appears more reasonable but produces way too much output
+            # in a normal Spack run
+            tty.debug(f"Attempting to read from missing scope: {self}")
```

Comment:
Since writes are generally always preceded by reads - a warning would IMO be the wrong choice anyhow (part of our intent here is to allow this as part of normal operation); you are already doing that so this is more a comment on the comment.

I think `str(self)` will output the name, but the path may also be useful in the debug message.

Reply:
Yeah good point.
Would we want `ConfigScope.__repr__` to report that as well?

Reply:
Ideally yes (but I don't think we don't consistently implement `repr`, or perhaps at all, so I'd say that's nice to have but not required here).

Reply:
We do happen to implement `__repr__` for `ConfigScope` so I'll add that there.